xDrip/Extensions/NibViewController.swift

```diff
@@ -42,3 +42,25 @@ extension NibLoadable where Self: UIViewController {
         return String(describing: self)
     }
 }
+
+extension NibLoadable where Self: UIView {
```

Comment:
Such approach would create a separate view and embed it into the current one. This is not a perfect approach. Use this class instead:
``` swift
open class NibView: UIView {
    public static func instantiate<T: UIView>() -> T {
        let nib = UINib(nibName: className, bundle: Bundle(for: self))
        guard let view = nib.instantiate(withOwner: nil, options: nil).first as? T else {
            fatalError("View: \(className) couldn't be instantiated from nib")
        }
        return view
    }
    
    override open func awakeAfter(using aDecoder: NSCoder) -> Any? {
        guard subviews.count == 0 else {
            return self
        }

        let currentClass = type(of: self)
        guard let view = Bundle(for: currentClass)
            .loadNibNamed(currentClass.className, owner: nil, options: nil)?
            .first as? UIView else {
            return self
        }

        view.frame = frame
        view.autoresizingMask = autoresizingMask
        view.translatesAutoresizingMaskIntoConstraints = translatesAutoresizingMaskIntoConstraints

        for constraint in constraints {
            var firstItem = constraint.firstItem
            if firstItem === (self as AnyObject?) {
                firstItem = view
            }
            var secondItem = constraint.secondItem
            if secondItem === (self as AnyObject?) {
                secondItem = view
            }

            if let firstItem = firstItem {
                view.addConstraint(NSLayoutConstraint(item: firstItem,
                                                      attribute: constraint.firstAttribute,
                                                      relatedBy: constraint.relation,
                                                      toItem: secondItem,
                                                      attribute: constraint.secondAttribute,
                                                      multiplier: constraint.multiplier,
                                                      constant: constraint.constant))
            }
        }

        return view
    }
}
```